compiler/src/typed/typed_well_formedness.re

```diff
@@ -289,42 +289,62 @@ module WellFormednessArg: TypedtreeIter.IteratorArgument = {
           _,
           [
             (
-              Unlabeled,
+              _,
               {
                 exp_desc:
                   TExpConstant(
                     Const_number(
-                      (Const_number_int(_) | Const_number_float(_)) as n,
+                      (
+                        Const_number_int(_) | Const_number_float(_) |
+                        Const_number_rational(_) |
+                        Const_number_bigint(_)
+                      ) as n,
                     ),
                   ),
               },
             ),
           ],
         )
           when
-            modname == "Int32"
+            modname == "Int8"
+            || modname == "Int16"
+            || modname == "Int32"
             || modname == "Int64"
+            || modname == "Uint8"
+            || modname == "Uint16"
             || modname == "Uint32"
             || modname == "Uint64"
             || modname == "Float32"
-            || modname == "Float64" =>
+            || modname == "Float64"
+            || modname == "Rational"
+            || modname == "BigInt" =>
         // NOTE: Due to type-checking, we shouldn't need to worry about ending up with a FloatXX value and a Const_number_float
         let n_str =
           switch (n) {
           | Const_number_int(nint) => Int64.to_string(nint)
           | Const_number_float(nfloat) => Float.to_string(nfloat)
-          | _ => failwith("Impossible")
+          | Const_number_rational({rational_num_rep, rational_den_rep}) =>
+            Printf.sprintf("%s/%s", rational_num_rep, rational_den_rep)
+          | Const_number_bigint({bigint_rep}) => bigint_rep
```

Comment:
Is this exhaustive? If so, you can streamline the first pattern match

Reply:
What do you mean by that?

Reply:
On lines 297-301 you are pattern matching and then assigning with `as n` but if this is exhaustive, those can just use `n` in the match. I'll make a suggestion above